core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -228,6 +261,46 @@ private[spark] class EventLoggingListener(
     fileSystem.rename(new Path(logPath + IN_PROGRESS), target)
   }
 
+  /**
+   * According to the updated event to modify the maintained event's metrics
+   * @param executorId  the executor whose metrics will be modified
+   */
+  private def updateModifiedMetrics(executorId: String): Unit = {
+    val toBeModifiedEvent = executorIdToModifiedMaxMetrics.get(executorId)
+    val latestEvent = executorIdToLatestMetrics.get(executorId)
+    toBeModifiedEvent match {
+      case None => if (latestEvent.isDefined) executorIdToModifiedMaxMetrics.update(
+        executorId, latestEvent.get)
```

Comment:
@squito , I'm sorry I made a mistake here, you are really careful. Thank you correct me again. You are right that `executorIdToLatestMetrics` will be updated before we call `updateModifiedMetrics`, I'll remove the `if` statement. Sorry to mislead you.